Yank/yamlbuild.py

```diff
@@ -1177,14 +1178,17 @@ class YamlBuilder:
     def yank_options(self):
         return self._isolate_yank_options(self.options)
 
-    def __init__(self, yaml_source=None):
+    def __init__(self, yaml_source=None, platform_name=None):
```

Comment:
That may be, but it just feels like a very bad habit to start in on. What do we do when we run into another location dependent option? I don't want to get just keep adding keywords if we can help it. 

There may be a more fundamental paradigm shift here we may want to talk about. Each of the YANK components are getting complex enough that trying to run them through a pure api or command line may soon be come untenable. e.g. the `yank prepare` CLI is nearly unusable already. I think we may want to consider burying some internals from users going forward. But this is a larger discussion to have in a different place